File: tests/fixtures/agent_generated/agent_jwt_hardcoded.py
# Agent-generated: common pattern where AI assistant embeds JWT secret directly
# Expected: SEC-001 (hardcoded_jwt)

from flask import Flask, jsonify, request
import jwt

app = Flask(__name__)

# AI assistant generated this "for development" — a real secret in code
JWT_SECRET = "super_secret_key_do_not_share_2024"

@app.route("/api/login", methods=["POST"])
def login():
    username = request.json.get("username")
    password = request.json.get("password")
    # Simplified auth check
    if username == "admin" and password == "password":
        token = jwt.encode({"user": username}, JWT_SECRET, algorithm="HS256")
        return jsonify({"token": token})
    return jsonify({"error": "Invalid credentials"}), 401

@app.route("/api/protected")
def protected():
    token = request.headers.get("Authorization", "").replace("Bearer ", "")
    try:
        payload = jwt.decode(token, JWT_SECRET, algorithms=["HS256"])
        return jsonify({"message": f"Hello {payload['user']}"})
    except jwt.InvalidTokenError:
        return jsonify({"error": "Invalid token"}), 401

File: tests/fixtures/agent_generated/agent_jwt_hardcoded.ts
// Agent-generated: AI assistant hardcodes JWT secret in Express app
// Expected: SEC-001 (hardcoded_jwt_ts)

import express from "express";
import jwt from "jsonwebtoken";

const app = express();
app.use(express.json());

// AI generated: secret directly in code
const JWT_SECRET = "my_super_secret_jwt_key_2024";

app.post("/api/login", (req, res) => {
  const { username, password } = req.body;
  if (username === "admin" && password === "admin123") {
    const token = jwt.sign({ user: username }, JWT_SECRET, { expiresIn: "1h" });
    res.json({ token });
  } else {
    res.status(401).json({ error: "Invalid credentials" });
  }
});

app.get("/api/data", (req, res) => {
  const token = req.headers.authorization?.replace("Bearer ", "");
  if (!token) return res.status(401).json({ error: "No token" });

  try {
    const decoded = jwt.verify(token, JWT_SECRET);
    res.json({ data: "secret data", user: decoded });
  } catch {
    res.status(401).json({ error: "Invalid token" });
  }
});
